File: test/style_tests/generator.js
function f(x) {
  return x + 1;
}
function* g() {
  yield f(1);
  yield f(2);
  return (yield f(3));
}
let h = g();
console.log(h.next().value);
console.log(h.next().value);
console.log(h.next().value);
console.log(h.next().value);

File: test/style_tests/generator.scala
def f(x : Int) =
  x + 1
gen def g =
  println("started")
  yield f(1)
  println("yielded once")
  yield f(2)
  println("yielded twice")
  yield f(3)
  println("yielded third times and end")
val h = g
println(next(h))
println(next(h))
println(next(h))
println(next(h))